lib/core/src/display/shape/text/text_field/word_occurrence.rs

```diff
@@ -142,25 +142,65 @@ impl WordOccurrences {
 
 
 
-// =============
-// === Words ===
-// =============
+// ===================
+// === IndexedChar ===
+// ===================
+
+/// A character represention with the index of its occurrence in text.
+#[derive(Debug,Clone,Copy)]
+pub struct IndexedChar {
+    #[allow(missing_docs)]
+    pub index     : usize,
+    #[allow(missing_docs)]
+    pub character : char
+}
+
+
+
+// ===================
+// === IndexedWord ===
+// ===================
+
+/// A word with indexed characters.
+#[derive(Debug,Shrinkwrap,Clone)]
+pub struct IndexedWord {
+    /// A collection of IndexedChar.
+    pub indexed_chars:Vec<IndexedChar>
```

Comment:
I didn't noticed it earlier, but perhaps it is enough to keep index of first character, and then the word?